### ISSUE TITLE
Is it possible to get something like this running in melonDS?

### ISSUE THREAD

**Issue by Wild-Dog** [melonDS-emu/melonDS#2236]
https://github.com/LRFLEW/AM64DS_DeSmuME

Being able to use a joystick for the camera would be amazing. Even if it's just for a mod like this, is it possible to do it in MelonDS?

**Comment by Jaklyy** [comment]
the exact implementation in this mod uses a custom gba cart slot, which should be fully doable in melonds without too much work. though it would be weird to add a feature that, to my knowledge, only one mod uses?

there is also another idea floating around in my head of trying to get those really cool 3ds compatible mods running in melonds, since those actually have real hardware to back them, and a bunch of mods exist for it. unfortunately i dont think it's realistic, since it's a lot of completely arbitrary code being executed on the arm11, and we dont really want to have to emulate the entire arm 11 just for one enhancement feature?

There is a more feasible approach, which unfortunately comes at the cost of creating our own standard for expanded control mods, where we add a lua scripting engine that would allow for directly sending the extra controller inputs into the game state in whatever way works best for the given mod.

**Comment by vitor251093** [comment]
I've made some experiments adding a camera joystick to a few games, so I figured I could share my two cents. Specifically, I would like to talk about games that allow you to control the camera using the DS touch screen, which are the most feasible ones.

At first, I thought a system for that would basically consist of starting with a touch in the center on the touch screen, and then moving the stylus in a way that mimics the expected behavior of the analog stick. So if you hold left it continuously moves to the left, until it reaches the corner, resetting the movement to the center of the touch screen. The same applies to all 4 directions, and those can be combined. Depending on how much you incline the analog stick, it moves the stylus in a faster or slower pace.

With Kingdom Hearts 358/2 Days, you can control the camera in a good enough manner if you do that, BUT once you return to the center of the touch screen, the camera moves back during a few frames, causing some "hiccups". However, if, instead of resetting, you hold the stylus at the corner of the screen, the movement is smooth with no issues.

With Kingdom Hearts Re:Coded however, if you hold the stylus at the corner, the camera just stops moving This leads to the fact that every game that implements a touch screen camera does have its own implementation. This specific situation could be sorted out with a checkbox though.

And then you have Metroid: Prime Hunters, which does require resetting on the corner, but if you move the stylus at the same pacing, it just won't move the camera. You need to increase the sensitivity of the analog stick to make it move properly.

**In conclusion:** I think this is feasible for games that allow you to control the camera using the touch screen, but it will require more than button inputs to be properly configured. At the very least, a "Sensitivity" field and a "Reset on corner" checkbox would be necessary, but that's just based on my experience with those three games.

I can open a pull request with the features I just mentioned (if the melonDS team is interested in them), but it may be necessary to test that with more games to be sure that it will usable for other games as well, like Super Mario 64 DS.

**Comment by vitor251093** [comment]
Just realized this is related to #2238 , so I figured it would be interesting to link the two of them.

Also, the cursor/joystick modes also makes sense.
